go/libkb/api_stub.go

```diff
@@ -5,31 +5,146 @@
 
 package libkb
 
+import "fmt"
+
 type StubAPIEngine struct {
 	*ExternalAPIEngine
+	mocking       bool
+	expectingJSON map[string]*ExternalAPIRes
+	expectingHTML map[string]*ExternalHTMLRes
+	expectingText map[string]*ExternalTextRes
+	calls         []stubAPIEngineCallRecord
+}
+
+type stubAPIEngineCallRecord struct {
+	kind     XAPIResType
+	endpoint string
 }
 
 func NewStubAPIEngine() *StubAPIEngine {
 	return &StubAPIEngine{
 		ExternalAPIEngine: &ExternalAPIEngine{BaseAPIEngine{clients: make(map[int]*Client)}},
+		mocking:           false,
+		expectingJSON:     make(map[string]*ExternalAPIRes),
```

Comment:
would id make more sense to have one map that maps a string a struct that has 3 fields?